HelpSource/Classes/Pgate.schelp

```diff
@@ -1,15 +1,40 @@
 class:: Pgate
-summary:: gated stream advances only when an event key is true
+summary:: A gated stream that only advances when a particular event key is true.
 related:: Classes/Pn
 categories:: Streams-Patterns-Events>Patterns>Repetition
-
 description::
 
-Pgate advances its subpattern whenever strong::key:: is true. Pgate must be used within an Event pattern.
+Pgate advances its subpattern whenever strong::key:: is true. Pgate must be used within an Event pattern. You can either set the key manually, or use LINK::Classes/Pn::.
+
+CLASSMETHODS::
+
+METHOD:: new
+
+ARGUMENT:: pattern
+The source pattern to be filtered.
+
+ARGUMENT::repeats
+Repeat the enclosed pattern strong::repeats:: times.
+
+ARGUMENT::key
+Pgate will only return the next value in the source pattern if the event that it belongs to has strong::key:: set to true. Otherwise it will keep returning the same value. This allows for a gated effect on streams.
 
 Examples::
 
 code::
+// Use \step to change the octave randomly
+(
+Pbind(
+
+	\degree,	Pseq((0..7), inf),
+	\step, Pseq([false, false, false, true, false, true, false], inf),
+	\octave,	Pgate(Pwhite(3,7), inf, \step),
+	\dur, 0.2
```

Comment:
sorry, i didn't notice this before, but the lines above have tabs after the commas instead of spaces for some reason, between `\degree,` and `Pseq`, and `\octave,` and `Pgate`. I can't tell if the other lines have tabs that are conveniently at the right tab stop to look like spaces. could you please fix that?

Reply:
I think those were in the original, because my example was a cut and paste job.

Should be fixed now.

Reply:
yeah, i believe i've seen that before. thanks for taking care of it :)